sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -560,18 +560,39 @@ private String parentResourceLinkToQueryLink(String parentResouceLink, ResourceT
     }
 
     private <T extends Resource> Flux<FeedResponse<T>> createQuery(
-            String parentResourceLink,
+        String parentResourceLink,
+        SqlQuerySpec sqlQuery,
+        CosmosQueryRequestOptions options,
+        Class<T> klass,
+        ResourceType resourceTypeEnum) {
+
+        String resourceLink = parentResourceLinkToQueryLink(parentResourceLink, resourceTypeEnum);
+        UUID activityId = Utils.randomUUID();
+        IDocumentQueryClient queryClient = documentQueryClientImpl(RxDocumentClientImpl.this);
+
+        InvalidPartitionExceptionRetryPolicy invalidPartitionExceptionRetryPolicy = new InvalidPartitionExceptionRetryPolicy(
```

Comment:
Not sure I picked up the best places to wire up the invalid partitionExceptionRetryPolicy. Several thoughts and questions:
1. Trying to put this logic as low as the query pipeline
2. Since for parallelQuery, each partition will have its own request, so at this point, there will be no request associate with this retry policy. 
3. I see usually we define next policy, but in this case, I do not think we need to define any. does it make sense?
4. For default document context, it already wired up InvalidPartitionExecptionRetry, but I do not see harm to wire it again here
5. Currently, for pipelinedDocumentQuery, we have retry policy for each partition. For example throttleRetryPolicy,  does it mean each partition will retry n times? should this number be shared across all the partitions?